src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/IdamConsumerTest.java

```diff
@@ -0,0 +1,191 @@
+package uk.gov.hmcts.reform.wataskmanagementapi;
+
+import au.com.dius.pact.consumer.MockServer;
+import au.com.dius.pact.consumer.Pact;
+import au.com.dius.pact.consumer.dsl.PactDslJsonBody;
+import au.com.dius.pact.consumer.dsl.PactDslJsonRootValue;
+import au.com.dius.pact.consumer.dsl.PactDslWithProvider;
+import au.com.dius.pact.consumer.junit5.PactConsumerTestExt;
+import au.com.dius.pact.consumer.junit5.PactTestFor;
+import au.com.dius.pact.model.RequestResponsePact;
+import com.google.common.collect.Maps;
+import io.restassured.http.ContentType;
+import lombok.extern.slf4j.Slf4j;
+import net.serenitybdd.rest.SerenityRest;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Slf4j
+@ExtendWith(PactConsumerTestExt.class)
+@ExtendWith(SpringExtension.class)
+public class IdamConsumerTest {
+
+    private static final String BEARER_TOKEN = "Bearer eyJ0eXAiOiJKV1QiLCJraWQiOiJiL082T3ZWdeRre";
+    private static final String IDAM_OPENID_TOKEN_URL = "/o/token";
+    private static final String IDAM_OPENID_USERINFO_URL = "/o/userinfo";
+
+    private static final String PACT_TEST_EMAIL_VALUE = "ia-caseofficer@fake.hmcts.net";
+    private static final String PACT_TEST_PASSWORD_VALUE = "London01";
+    private static final String PACT_TEST_CLIENT_ID_VALUE = "pact";
+    private static final String PACT_TEST_CLIENT_SECRET_VALUE = "pactsecret";
+    private static final String PACT_TEST_SCOPES_VALUE = "openid profile roles";
+    private static final String PACT_TEST_ROLES_VALUE = "caseworker";
+
+    @Pact(provider = "Idam_api", consumer = "wa_task_management_api")
```

Comment:
It appears we are referring to an out of date implementation of case-api. I spoke to Satya and we should use the implementation on the following ia-case-api branch  "ConsumerContractTests".

For example;
In @Pact(provider = "Idam_api", consumer = "wa_task_management_api"), Idam_api should be idamApi_oidc